doc/compiler/MemoryManager.md

```diff
@@ -0,0 +1,226 @@
+managing its own memory for the same reasons as the OMR Compiler. 
+However, in order to ensure that all memory in the JVM is accounted 
+for, as well as to minimize the likelihood of memory leaks,
+the OpenJ9 Compiler Memory Manager uses JVM APIs instead 
+of OS APIs directly.
+
+Some parts of the memory manager override the OMR implementations, while 
+others either extend or directly use them. For information on any terms 
+not documented here, refer to the documentation found in 
+[eclipse/omr](https://github.com/eclipse/omr/blob/master/doc/compiler/memory/MemoryManager.md).
+
+## Compiler Memory Manager Hierarchy
+
+### Low Level Allocators
+Low Level Allocators cannot be used by STL containers. They are 
+generally used by High Level Allocators as the underlying 
+provider of memory.
+
+```
+                          +---------------------+
+                          |                     |
+                          | TR::SegmentProvider |
+                          |                     |
+                          +-----+-----------+---+
+                                ^           ^
+                                |           |
+                                |           |
+                                |           |
+          +---------------------+--+     +--+-----------------------+
+          |                        |     |                          |
+          |  TR::SegmentAllocator  |     | J9::DebugSegmentProvider |
+          |                        |     |                          |
+          +--------+---------------+     +--------------------------+
+                   ^
+                   |
+                   |
++------------------+---------+
+|                            |
+| J9::SystemSegmentProvider  |
+|                            |
++----------------------------+
+
+
+--------------------------------------------------------------------
+
+
+                +---------------------+
+                |                     |
+                | J9::SegmentProvider |
+                |                     |
+                +-----+-----------+---+
+                      ^           ^
+                      |           |
+                      |           |
+                      |           |
++---------------------+--+     +--+---------------+
+|                        |     |                  |
+|  J9::SegmentAllocator  |     | J9::SegmentCache |
+|                        |     |                  |
++------------------------+     +------------------+
+
+```
+
+#### J9::SystemSegmentProvider
+`J9::SystemSegmentProvider` implements `TR::SegmentAllocator`. 
+It uses `J9::SegmentProvider` (see below) in order to allocate 
+`TR::MemorySegment`s and uses `TR::RawAllocator`for all its 
+internal memory management requirements. As part of its initialization, it
+preallocates a segment of memory. This is the default Low 
+Level Allocator used by the Compiler. However, the compiler will
```

Comment:
I wasn't 100% clear on this last sentence.  I think its clearer to say "... the compiler will use `OMR::DebugSegmentProvider` instead of `J9::SegmentProvider` when..."